diff --git a/src/freedreno/vulkan/tu_cmd_buffer.cc b/src/freedreno/vulkan/tu_cmd_buffer.cc
index 3f0365a9..087d5122 100644
--- a/src/freedreno/vulkan/tu_cmd_buffer.cc
+++ b/src/freedreno/vulkan/tu_cmd_buffer.cc
@@ -3217,6 +3217,7 @@ tu7_emit_concurrent_binning_start(struct tu_cmd_buffer *cmd,
           cmd->state.lrz.image_view &&
           !fdl6_lrz_fc_fully_covered(&cmd->state.lrz.image_view->image->lrz_layout),
           cmd, "partial LRZ fast clear") ||
+       tu7_cb_disable_reason(cmd->state.rp.lrz_rmw_emitted, cmd, "LRZ RMW in render pass") ||
        tu7_cb_disable_reason(!cmd->device->instance->drirc.perf.allow_concurrent_binning, cmd,
                              "globally disabled")) {
      tu_cs_emit_pkt7(cs, CP_THREAD_CONTROL, 1);
@@ -6371,6 +6372,7 @@ tu_render_pass_state_merge(struct tu_render_pass_state *dst,
    dst->disable_gmem |= src->disable_gmem;
    dst->sysmem_single_prim_mode |= src->sysmem_single_prim_mode;
    dst->lrz_disable_for_next_rp |= src->lrz_disable_for_next_rp;
+   dst->lrz_rmw_emitted |= src->lrz_rmw_emitted;
    dst->draw_cs_writes_to_cond_pred |= src->draw_cs_writes_to_cond_pred;
    dst->shared_viewport |= src->shared_viewport;
    dst->read_only_input_attachments |= src->read_only_input_attachments;
diff --git a/src/freedreno/vulkan/tu_cmd_buffer.h b/src/freedreno/vulkan/tu_cmd_buffer.h
index 4b700900..40233bb2 100644
--- a/src/freedreno/vulkan/tu_cmd_buffer.h
+++ b/src/freedreno/vulkan/tu_cmd_buffer.h
@@ -318,6 +318,7 @@ struct tu_render_pass_state
    bool disable_gmem;
    bool sysmem_single_prim_mode;
    bool lrz_disable_for_next_rp;
+   bool lrz_rmw_emitted;
 
    /* This is set if, at any point in the render pass, we were not able to
     * duplicate the viewport per-view due to the user using multiple viewports
diff --git a/src/freedreno/vulkan/tu_lrz.cc b/src/freedreno/vulkan/tu_lrz.cc
index 15e1eca4..b8fe25ad 100644
--- a/src/freedreno/vulkan/tu_lrz.cc
+++ b/src/freedreno/vulkan/tu_lrz.cc
@@ -206,6 +206,7 @@ tu_lrz_emit_force_disable_for_rp(struct tu_cmd_buffer *cmd, struct tu_cs *cs)
    if (CHIP >= A7XX) {
       const struct tu_reg_value reg = GRAS_SC_BIN_CNTL(CHIP, .force_lrz_dis = true);
 
+      cmd->state.rp.lrz_rmw_emitted = true;
       cs->rmw(reg, { .src0 = ~0u, .src1 = reg.value });
    } else {
       /* A6XX does not support GRAS_SC_BIN_CNTL.FORCE_LRZ_DIS */
@@ -1146,6 +1147,7 @@ tu_lrz_flush_valid_at_secondary_rp_boundary(
          tu_lrz_disable_write_for_rp(cmd, reason);
       }
 
+      cmd->state.rp.lrz_rmw_emitted = true;
       tu_lrz_emit_disable_write_for_rp<CHIP>(cs);
       return;
    }
@@ -1163,6 +1165,7 @@ tu_lrz_flush_valid_at_suspending_rp_boundary(struct tu_cmd_buffer *cmd,
       return;
 
    if (cmd->state.lrz.valid) {
+      cmd->state.rp.lrz_rmw_emitted = true;
       tu_lrz_emit_disable_write_for_rp<CHIP>(cs);
       return;
    }
